src/content/blog/please-stop-sending-me-nested-dependency-security-reports/index.mdx

```diff
@@ -0,0 +1,97 @@
+---
+description: "Security is important, but overusing report mechanisms for non-actionable noise is counterproductive."
+image:
+  alt: Larry David in Curb Your Enthusiasm looking consternated
+  src: larry-david-consternation-head.webp
+pubDate: 2024-04-29
+title: "Please Stop Sending Me Nested Dependency Security Reports"
+---
+
+import LabeledVideo from "~/components/blog/mdx/LabeledVideo.astro";
+
+As an open source maintainer, I regularly receive "critical security vulnerability" DMs, emails, and GitHib issues about my projects.
+Most of these notices are sent from individual developers or IT teams about packages that are dependencies of dependencies.
+These "nested dependency" reports often go something like:
+
+<blockquote>
+Hello.
+We use `your-package` and `npm audit` is reporting a critical issue:
+<br />
+```plaintext
+some-nested-dependency  <=0.1.2
+Severity: critical
+Prototype Pollution in some-nested-dependency - https://github.com/advisories/GHSA-ab12-cd23-ef45
+No fix available
+node_modules/some-dependency/node_modules/some-nested-dependency
+  some-dependency  >=2.3.4
+  Depends on vulnerable versions of some-nested-dependency
+  node_modules/some-dependency
+    your-package  >=5.6.7
+    Depends on vulnerable versions of some-dependency
+    node_modules/your-package
+```
+<br />
+Could you please fix this?
+</blockquote>
+
+I appreciate the attention to detail and security.
+Really, I do!
+Paying attention to your tooling is a useful skill for any developer.
```

Comment:
optional joke: "And I know that software toolchain vulnerabilities really matter. Trust me, I'm not trying to create SolarWinds 2.0 here."

Reply:
Ha, thanks. Tempting... I'm going to omit as most of my target audience won't remember what that is, and I'm hoping for a lean post. But thanks for the reminder 😄 what a fascinating attack.